src/peer/negotiator.js

```diff
@@ -477,6 +438,77 @@ class Negotiator extends EventEmitter {
   }
 
   /**
+   * Replace the stream being sent with a new one.
+   * @param {MediaStream} newStream - The stream to replace the old stream with.
+   * @private
+   */
+  _replacePerTrack(newStream) {
+    const vTracks = newStream.getVideoTracks();
+    const aTracks = newStream.getAudioTracks();
+    let vSenders = [];
+    let aSenders = [];
+    this._pc.getSenders().forEach(sender => {
+      sender.track.kind === 'video' && vSenders.push(sender);
+      sender.track.kind === 'audio' && aSenders.push(sender);
+    });
+
+    // no video tracks in newStream => remove all RtpSender(.kind = video)
+    // otherwise, replace it or create it
+    // We assume that there is at most 1 audio stream and at most 1 video stream in localStreams
+    if (vTracks.length === 0) {
+      vSenders.forEach(sender => this._pc.removeTrack(sender));
+    } else {
+      vTracks.forEach((track, idx) => {
+        if (vSenders.length === 0) {
+          // no video track in previous stream, so we can't replace it
+          this._pc.addTrack(track, newStream);
+        } else {
+          // need to check id before replace..?
+          vSenders[idx].replaceTrack(track);
+        }
+      });
+    }
+
+    // do the same thing for audio tracks
+    if (aTracks.length === 0) {
+      aSenders.forEach(sender => this._pc.removeTrack(sender));
+    } else {
+      aTracks.forEach((track, idx) => {
+        if (aSenders.length === 0) {
+          // no audio track in previous stream, so we can't replace it
+          this._pc.addTrack(track, newStream);
+        } else {
+          aSenders[idx].replaceTrack(track);
+        }
+      });
+    }
+
+    this._replaceStreamCalled = true;
+
+    // We don't actually need to do renegotiation but force it in order to prevent
+    // problems with the stream.id being mismatched when renegotiation happens anyways
+    // Firefox 55 doesn't trigger onnegotiationneeded event when a track in a stream is removed
+    this._pc.onnegotiationneeded();
+  }
+
+  /**
+   * Replace the stream being sent with a new one.
+   * @param {MediaStream} newStream - The stream to replace the old stream with.
+   * @private
+   */
+  _replacePerStream(newStream) {
+    const localStreams = this._pc.getLocalStreams();
+    // We assume that there is at most 1 stream in localStreams
+    if (localStreams && localStreams[0]) {
+      this._pc.removeStream(localStreams[0]);
+    }
+
+    this._replaceStreamCalled = true;
```

Comment:
This line should be placed after `this._pc.addStream(newStream);` for naming..?

Reply:
`_replaceStreamCalled` is true after `replaceStream` and is false after `onnegotiationneeded`.
`_replaceStreamCalled` is used to check if `onnegotiationneeded` event has finished or not. 
If it is placed after `addStream`, `_replaceStreamCalled` becomes true after `onnegotiationneeded`.